### PR TITLE
feat: emit batch

### PR DESCRIPTION
**Pull request by gunb0s**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #12748

## What is the new behavior?

enable kafka producer to emit batch

## Does this PR introduce a breaking change?
- [v] Yes
- [ ] No

## Other information

### CODE REVIEW

File: packages/microservices/client/client-kafka.ts [hunk 1]
```diff
@@ -223,6 +232,56 @@ export class ClientKafka extends ClientProxy {
...
+      resetOnDisconnect: false,
+    });
+    connectableSource.connect();
+    return connectableSource;
+  }
+
+  async dispatchBatchEvent<TInput = any>(
+    packets: ReadPacket<{ messages: TInput[] }>,
+  ): Promise<any> {
+    if (packets.data.messages.length === 0) return;
```
Comment (Tony133) [inline packages/microservices/client/client-kafka.ts:256]:
```suggestion
    if (packets.data.messages.length === 0) {
       return;
    }
```

**Review by Tony133 — CHANGES_REQUESTED** [review]
For the rest it seems ok 👍

File: packages/microservices/client/client-kafka.ts (hunk 1 above)
Comment (gunb0s) [reply packages/microservices/client/client-kafka.ts]: [feat: add block wrapping return](https://github.com/nestjs/nest/pull/12954/commits/cb8e04dd62d85f37b61936f53a50f1bb8169e52c)

**Review by Tony133 — APPROVED** [review]

**Comment by apparelmagic-heath** [comment]
Would love to see this feature merged. Very handy for our use-case.

**Comment by bozorgmehr96** [comment]
Is this going to be merged, and if so, do you have any estimates for when?

And another thing, are you open to batching now? You said no here, #9455
Since you green-lighted batch producing, I need batch consuming and want to know if I should also have hope for it. I might start a PR for it if I find some free time.

**Comment by binh-truong-tpv-mti** [comment]
hi all, pls consider this PR

**Comment by posquit0** [comment]
@Tony133 @kamilmysliwiec I need this feature, too! Could you consider to merge this PR?

**Comment by Tony133** [comment]
> @Tony133 @kamilmysliwiec I need this feature, too! Could you consider to merge this PR?

It need to be seen whether it can be merged into version 10.x of NestJS or put into the future version 11.

**Pull request merged**